### ISSUE TITLE
Datastore: Client requires authentication even when connecting to the datastore emulator

### ISSUE THREAD

**Issue by rylz** [googleapis/python-datastore#11]
When I try to construct a datastore client on a box that has the datastore emulator enabled and running, I get the following:
```
  File ".../model/datastore.py", line 6, in <module>
    client = datastore.Client()
  File ".../lib/python3.6/site-packages/google/cloud/datastore/client.py", line 205, in __init__
    project=project, credentials=credentials, _http=_http)
  File ".../lib/python3.6/site-packages/google/cloud/client.py", line 215, in __init__
    _ClientProjectMixin.__init__(self, project=project)
  File ".../lib/python3.6/site-packages/google/cloud/client.py", line 169, in __init__
    project = self._determine_default(project)
  File ".../lib/python3.6/site-packages/google/cloud/datastore/client.py", line 222, in _determine_default
    return _determine_default_project(project)
  File ".../lib/python3.6/site-packages/google/cloud/datastore/client.py", line 75, in _determine_default_project
    project = _base_default_project(project=project)
  File ".../lib/python3.6/site-packages/google/cloud/_helpers.py", line 179, in _determine_default_project
    _, project = google.auth.default()
  File ".../lib/python3.6/site-packages/google/auth/_default.py", line 283, in default
    raise exceptions.DefaultCredentialsError(_HELP_MESSAGE)
google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or
explicitly create credential and re-run the application. For more
```

However, it's clear that these credentials are not actually required because when I initialize my client as follows, it works in both dev and production:
```
if os.environ.get(environment_vars.GCD_HOST):
    # avoid a bug in the google client - it tries to authenticate even when the emulator is enabled
    client = datastore.Client(_http=requests.Session)
else:
    client = datastore.Client()
```

With that change, the library works fine for me, but this seems like something the library ought to be able to detect and handle by default to save devs a lot of time.

In case they're relevant:

2. OS type and version
```
$ uname -a
Darwin Rileys-MacBook-Pro.local 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64
```

3. Python version and virtual environment information `python --version`
```
$ python --version
Python 3.6.5
```
4. google-cloud-python version `pip show google-cloud`, `pip show google-<service>` or `pip freeze`
```
$ pip show google-cloud-datastore
Name: google-cloud-datastore
Version: 1.7.0
```

**Comment by HemangChothani** [comment]
Closing this issue, because it has been resolved.
It's working fine without passing `credentials=AnonymousCredentials()` as parameter. 

feel free to reopen if needed!

**Comment by c4rlo** [comment]
I'm experiencing this issue with `google-cloud-datastore` 1.14.0.

It only works with `credentials=AnonymousCredentials()`.

In addition to that I found I needed to export `GOOGLE_CLOUD_PROJECT`; passing `project=` did not work.

**Comment by c4rlo** [comment]
> feel free to reopen if needed!

I would, but lack privileges to do so.

**Comment by HemangChothani** [comment]
@c4rlo Cloud you please share `google-cloud-sdk` version,  i am not able  to reproduce it.

Here are the steps:

1. Start datastore emulator using gcloud sdk.
2. Set the env variable : DATASTORE_EMULATOR_HOST
3. import datastore and create client

```python 
from google.cloud import datastore
from google.auth.credentials import AnonymousCredentials

client = datastore.Client(
    project="emulated-project",
    namespace='ns_test',
    credentials=AnonymousCredentials(),
    _http=None
)

**Comment by tseaver** [comment]
In addition to passing an instance of `AnonymousCredentials()`, you could also use the `EmulatorCreds` class from `google-test-utils`, e.g.:

```bash
/path/to/venv/bin/pip install google-test-utils
```

```python
from google.cloud import datastore
from test_utils.system import EmulatorCreds

client = datastore.Client(
    project="emulated-project",
    namespace='ns_test',
    credentials=EmulatorCreds(),
    _http=None
)
```

**Comment by c4rlo** [comment]
I use `google-cloud-sdk` version 1.14.0, currently the latest.

As I mentioned, when explicitly passing `credentials=AnonymousCredentials()`, it works fine (disregarding the `project=` issue for now).

However, I don't see a requirement for that documented anywhere. It's not mentioned either in the [emulator docs](https://cloud.google.com/datastore/docs/tools/datastore-emulator) or in the [`google-cloud-datastore` docs](https://googleapis.dev/python/datastore/latest/index.html), as far as I can tell. How are users supposed to find out about this?

Moreover, why should this even be necessary? As the original comment in this issue said:

> this seems like something the library ought to be able to detect and handle by default to save devs a lot of time

**Comment by HemangChothani** [comment]
@c4rlo for emulator you can use `export DATASTORE_PROJECT_ID=my-project-id` to declare the project or you can pass the project as a parameter.

**Comment by c4rlo** [comment]
Ok, `export DATASTORE_PROJECT_ID=my-project-id` does in fact work correctly for me now. Thanks!

However, that was not really the main issue.

The main issue is the need to pass `credentials=AnonymousCredentials()` (or I suppose `EmulatorCreds()` as mentioned). Why does the library not infer this automatically given I have `DATASTORE_EMULATOR_HOST` set? (Along with four other `DATASTORE_*` environment variables set by `$(gcloud beta emulators datastore env-init)`.)

And if there is a good reason for this, then why is the need for this not documented anywhere?

**Comment by HemangChothani** [comment]
> The main issue is the need to pass credentials=AnonymousCredentials() (or I suppose EmulatorCreds() as mentioned). Why does the library not infer this automatically given I have DATASTORE_EMULATOR_HOST set? 

@c4rlo I think it's new feature request to set at `EmulatorCreds` automatically if found `DATASTO RE_EMULATOR_HOST` is set.

@tseaver Your view on this, `python-spanner` has done the same thing like the user demands.

**Comment by c4rlo** [comment]
I see this is now fixed in 1.15.0 via #71. Many thanks, @tseaver!
